lib/xandra/cluster/control_connection.ex

```diff
@@ -226,10 +232,11 @@ defmodule Xandra.Cluster.ControlConnection do
         %__MODULE__{} = data
       ) do
     select_peers_query =
-      "SELECT peer, data_center, host_id, rack, release_version, schema_version, tokens FROM system.peers"
+      "SELECT peer, data_center, host_id, rack, release_version, rpc_address, schema_version, tokens FROM system.peers"
 
     with {:ok, peers} <- query(data, connected_node, select_peers_query),
-         host when not is_nil(host) <- Enum.find(peers, fn peer -> peer["peer"] == address end) do
+         host when not is_nil(host) <-
+           Enum.find(peers, fn peer -> peer["rpc_address"] == address end) do
```

Comment:
I think this probably needs to be configurable. I think the "normal" case would be to use `peer`, but I can't find any examples on the Internet where these two addresses are different. Except in production at Community...

I would put something like `peer_address_field` in the options, or similar. @whatyouhide may have a better idea of how to configure that.